Tests/UKHO.MaritimeSafetyInformation.Web.UnitTests/Controllers/WebhookControllerTests.cs

```diff
@@ -47,25 +49,115 @@ public void WhenValidHeaderRequestedInNewFilesPublishedOptions_ThenReturnsOkResp
                 HttpContext = context
             };
 
-            var result = (OkObjectResult)_controller.NewFilesPublishedOptions();
+            OkObjectResult result = (OkObjectResult)_controller.NewFilesPublishedOptions();
 
             Assert.AreEqual(200, result.StatusCode);
             Assert.AreEqual("*", _controller.HttpContext.Response.Headers.Where(a => a.Key == "WebHook-Allowed-Rate").Select(b => b.Value).FirstOrDefault());
             Assert.AreEqual(requestHeaderValue, _controller.HttpContext.Response.Headers.Where(a => a.Key == "WebHook-Allowed-Origin").Select(b => b.Value).FirstOrDefault());
         }
 
         [Test]
-        public async Task WhenNewFilesPublishedIsCalled_ThenReturnsOkResponse()
+        public async Task WhenNewFilesPublishedIsCalledWithValidData_ThenShouldReturnOkResponse()
+        {
+            string jsonString = GetRequestString();
+            MemoryStream requestData = new(Encoding.UTF8.GetBytes(jsonString));
```

Comment:
This code is repeated in all test cases, requestData can be set in Setup() and then reused in required test cases.

Reply:
Common code moved to Setup()